File: fleet-pay-export/api/src/models/contractor.ts
import { User } from './user';
import { Company } from './company';
import { Contact } from './contact';
import { FundingSetting } from './funding_setting';
import { PaymentStructure } from './payment_structure';

export interface Contractor {
  id: string,
  name: string,
  userId: string,
  user?: User | null,
  pendingInvoiceAmount: number,
  emails: string[],
  phoneNumbers: string[],
  companyId: string,
  company?: Company | null,
  primaryContact: Contact,
  fundingSetting: FundingSetting,
  paymentStructure: PaymentStructure,
  deleted: boolean,
  lasPaymentRequestDate?: Date | null,
  // --- start extra_fields code ---
  
  // --- end extra_fields code ---
}

File: 1-Future/3 fleet-pay-export/flutter/lib/models/contractor.dart
import 'package:freezed_annotation/freezed_annotation.dart';

import 'user.dart';
import 'company.dart';
import 'contact.dart';
import 'funding_setting.dart';
import 'payment_structure.dart';
// --- start import code ---
  
// --- end import code ---

part 'contractor.g.dart';
part 'contractor.freezed.dart';

@unfreezed
class Contractor with _$Contractor {
  // --- start init code ---
  
  // --- end init code ---

  factory Contractor({
    required String id,
    required String name,
    required String userId,
    User? user,
    required num pendingInvoiceAmount,
    required List<String> emails,
    required List<String> phoneNumbers,
    required String companyId,
    Company? company,
    required Contact primaryContact,
    required FundingSetting fundingSetting,
    required PaymentStructure paymentStructure,
    required bool deleted,
    DateTime? lasPaymentRequestDate,
  }) = _Contractor;

  factory Contractor.fromJson(Map<String, dynamic> json) => _$ContractorFromJson(json);

  // --- start extension code ---
  
  // --- end extension code ---

}
